boot/bootutil/src/loader.c

```diff
@@ -1039,6 +1040,12 @@ boot_go(struct boot_rsp *rsp)
 
     switch (swap_type) {
     case BOOT_SWAP_TYPE_NONE:
+#ifdef BOOTUTIL_VALIDATE_SLOT0
```

Comment:
Yeah, I think #ifdef is fine.  I just have a memory from long ago of getting "burned" by this.  We were using a config system similar to Mynewt's (header file full of #defines).  I wanted to be sure a feature was disabled, so I defined the corresponding symbol to 0.  Because the code checked if the macro was defined, rather than its actual value, I actually ended up enabling the feature.

As you said, #ifdef is more common, so it is probably best to stick with that.